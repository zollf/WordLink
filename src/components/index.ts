export { default as MenuBar } from './MenuBar';
export { default as MenuItem } from './MenuItem';
export { default as Button } from './Button';
export { default as LevelButtons } from './LevelButtons';
export { default as LevelSelect } from './LevelSelect';
export { default as Levels } from './Levels';
export { default as InputField } from './InputField';
export { default as NewUserPage } from './NewUserPage';
export { default as CircleButton } from './CircleButton';
export { default as Game } from './Game';
export { default as RegisteredUserPage } from './RegisteredUserPage';
export { default as UserStats } from './UserStats';
export { default as UserPuzzleStats } from './UserPuzzleStats';
export { default as Logo } from './Logo';
export { default as CircleText } from './CircleText';
