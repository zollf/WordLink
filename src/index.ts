export { default as Index } from './index/index';
export { default as Menu } from './menu/index';
export { default as Welcome } from './welcome/index';
export { default as Profile } from './profile/index';
export { default as Game } from './game/index';
export { default as Settings } from './settings';
