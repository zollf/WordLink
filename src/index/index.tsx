import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const Index = () => {
  return (
    <View style={styles.container}>
      <Text>Open up App.tsx to start working on your app!</Text>
    </View>
  );
}

export default Index;