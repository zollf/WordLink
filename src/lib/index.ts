export { default as getLevel } from './getLevel';
