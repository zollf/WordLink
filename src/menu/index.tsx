import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const Menu = () => {
  return (
    <View style={styles.menu}>

    </View>
  );
};

export default Menu;