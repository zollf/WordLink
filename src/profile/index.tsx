import React from 'react';
import { Text, View } from 'react-native';

import styles from './styles';

const Profile = () => {
  return (
    <View style={styles.profile}>

    </View>
  );
};

export default Profile;