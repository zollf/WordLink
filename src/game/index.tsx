import React from 'react';
import { View } from 'react-native';
import { observer } from 'mobx-react';

import { Game as WordLink } from '../../components';

import styles from './styles';

const Game = () => (
  <View style={styles.game}>
    <WordLink />
  </View>
);

export default observer(Game);