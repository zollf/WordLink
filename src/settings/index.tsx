import React from 'react';
import { View } from 'react-native';
import { observer } from 'mobx-react';

import { useStore } from '../../hooks';
import { MenuBar, Button } from '../../components';

import styles from './styles';

const Settings = () => {
  const { setStartingPage, setCurrentPage } = useStore('global');

  const handleClick = () => {
    setCurrentPage('menu');
    setStartingPage(true);
  };

  return (
    <View style={styles.settings}>
      <Button text="Back" onPress={handleClick} />
      <MenuBar />
    </View>
  );
};

export default observer(Settings);
