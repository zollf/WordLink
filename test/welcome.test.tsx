import React from 'react';
import useWithStores from './useWithStores';

import { Welcome } from '../src';

import renderer from 'react-test-renderer';

const component = useWithStores(<Welcome />);

describe('<Welcome/>', () => {
  it('matches snapshot', () => {
    expect(renderer.create(component)).toMatchSnapshot();
  });
});