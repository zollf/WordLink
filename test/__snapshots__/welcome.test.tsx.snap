// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome/> matches snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Welcome Back!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "width": 450,
          }
        }
      >
        Username
      </Text>
      <TextInput
        allowFontScaling={true}
        data-test-id="text-input"
        editable={true}
        onChange={[Function]}
        rejectResponderTermination={true}
        selectTextOnFocus={true}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "height": 93.5,
              "shadowColor": "#000",
              "shadowRadius": 4,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "#40BAFF",
              "width": 450,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2.5,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294956160,
                  4294950464,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 26.5,
                }
              }
            >
              START
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
  <View
    data-test-id="wrapper"
    style={
      Object {
        "alignItems": "center",
        "bottom": 106.5,
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      data-test-id="menu-bar"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#C8C8C8",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "center",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
          "width": 212,
          "zIndex": -1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "height": 50,
              "position": "absolute",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
              "zIndex": -1,
            },
            Object {
              "left": undefined,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
